### ISSUE TITLE
Fix linting errors

### ISSUE THREAD

**Issue by tchamberlin** [GreenBankObservatory/dysh#76]
I uncovered some minor bugs/issues while I was setting up linting:

```sh
$ hatch run lint --ignore E501,PD901
benchmark/makeplots.py:266:14: F821 Undefined name `ax`
benchmark/makeplots.py:271:9: F821 Undefined name `ax`
src/dysh/_dysh_init.py:35:5: B018 Found useless expression. Either assign it to a variable or remove it.
src/dysh/_dysh_init.py:44:11: F821 Undefined name `version`
src/dysh/fits/sdfitsload.py:160:18: F821 Undefined name `SpectrumList`
src/dysh/fits/sdfitsload.py:188:21: B018 Found useless attribute access. Either assign it to a variable or remove it.
src/dysh/spectra/core.py:96:41: F821 Undefined name `pair`
src/dysh/spectra/core.py:117:29: B028 No explicit `stacklevel` keyword argument found
src/dysh/spectra/core.py:148:29: B028 No explicit `stacklevel` keyword argument found
src/dysh/util/core.py:63:12: F821 Undefined name `sys`
Found 10 errors.
```
I can take a stab at fixing them if you want, but figured I'd check first. There are also a lot of issues with long comment lines (E501). These won't trigger issues in pre-commit, but they should probably be addressed at some point -- long lines don't work very will when doing diffs, or using GitHub UI.

`hatch run lint` also checks for Pandas issues, which right now is just complaining about variables named `df` (E901). I don't think that's super useful, so I can just ignore that one. But there are other pitfalls with Pandas that I think it is good to check for

If you want to see _all_ linting issues, you can do the following on branch `main-run-precommit`:

```sh
hatch run lint-all
```

The vast majority of these aren't enabled by default, either because they are just kinda dumb, or they are very tedious to fix. But there might be some useful stuff in there

**Comment by tchamberlin** [comment]
I'm going to re-use this issue for bug fixes. I have a branch that:
- introduces `ruff` (replacement for pylint) linting as pre-commit hook
- Runs `ruff --fix` on all code

I have also fixed as many "obvious" bugs as I can, but there are still several that I'm reticent to change, so those are left unfixed right now. PR soon

**Comment by astrofle** [comment]
@tchamberlin can we close this issue?

**Comment by tchamberlin** [comment]
Yes most of this was addressed by #74. Although we seem to have regressed a bit (e.g. `hatch run lint --select F821` should never catch anything), and this should be addressed again in the future
